### DISCUSSION TITLE
Minimal example with `task sync init` provoking Could not find common ancestor

### DISCUSSION THREAD

**Discussion by bradyt** [GothenburgBitFactory/taskwarrior#2739] [General]
I think the following indicates a minimal broken example:

```
> task add foo
> task add bar
> task sync
> task sync init
Please confirm that you wish to upload all your tasks to the Taskserver (yes/no) yes
Syncing with localhost:53589

Sync failed.  The Taskserver returned error: 500 ERROR: Could not find common ancestor for 11111111-1111-1111-1111-111111111111
```

How this manifests itself in less trivial or artificial ways may take some imagination. One approach is to search GBF:

- https://github.com/search?q=org:GothenburgBitFactory+Could+not+find+common+ancestor+for&state=open&type=issues

We as users can do a lot of strange things. For example, we might troubleshoot our server setups, creating new taskd accounts, and point our existing client setups at the new taskd account. We get "client sync key not found", search the web, and determine to try `task sync init`. This might result in `Could not find common ancestor for`. This one can be a bit more mysterious than `client sync key not found`.

What I'm hoping is that someone can enlighten me on why, for the simple example above, I should expect to get the error about common ancestor. If we can narrow down each occurrence to an illustrative minimal example, and, understand why it happens, we as users, or anyone trying to support users, might be better able to understand issues as they arise.

If it's helpful, here is a link to the line in Taskserver where this error seems to be thrown:

> ```cpp
>    throw std::string ("ERROR: Could not find common ancestor for ") + uuid;
> ```
> - https://github.com/GothenburgBitFactory/taskserver/blob/v1.1.0/src/daemon.cpp#L642

**Comment by djmitche** [comment]
What is the rational for running `task sync init` after `task sync`?

**Reply by bradyt** [reply]
It may be equivalent to several scenarios, and some common steps advised on issue tracker and related projects.

For example, if user is running `task sync init` from more than one client.

Perhaps I should have resummarized, as my recent conclusion is that this is currently not supported by Taskserver, despite how often it appears in issue trackers.

But the example above may help to establish for anyone tempted to do something that is equivalent to this minimal example.

So this may serve to illustrate that this is unsupported.

**Reply by bradyt** [reply]
I think one of the motivations for people, was `task sync init` is similar to discarding a backlog.data, to get rid of a missing sync key, but not only sent updates that were discarded in the backlog.data, but even updates that were only on that client, but no longer was in that discarded backlog.data.

**Reply by bradyt** [reply]
And in general, the above is a conjecture, that many of the reports across this and related projects, where user has `Could not find common ancestor for`, if each user then reduced their issue to a minimal broken example, the above might be the minimal reproduce. So it could be useful for establishing lack of support, or if someone tries to fix any "actual" issue.
